figures/sync_architecture.xml

```diff
@@ -0,0 +1 @@
+<mxfile type="google" userAgent="Mozilla/5.0 (Macintosh; Intel Mac OS X 10_11_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/56.0.2924.87 Safari/537.36" version="6.3.1" editor="www.draw.io"><diagram name="Page-1">7Vtdc+I2FP01zLQPYSTLxvZjyGZ3H7YzTLKdtk8ZYQvbjbGoLBLIr69sS2BZsEuKTKAJL+CLvnx07rn3CjNAN/PVF4YX6W80JvnAAfFqgD4NHAcGPhBvlWUtLRD6jSVhWdzYwNZwn70Q2VBZl1lMSq0hpzTn2UIaYWOMaFGQiGs2zBh91pvNaB5rhgVOiGG4j3BuWv/IYp42Vi9wt/avJEtSObMbyEVOcfSYMLos5HQDB83qV/P1HKuhZPsyxTF9bpnQ7QDdMEp582m+uiF5hW0HtZKv1VIHaJzyeS4uoPhYf/15T2d4SGdxc4wUvD3dvvF83xiQxAJHeUkZT2lCC5zfbq3jGh1SjQD02f8mnK8lFfCSU2HajvCN0oVsJxbH1n9W/YduoK7/an85ISybE06YnKTkjD6SG5pTVq8Sgfq1844VRnTJInlPvrRxzBIim/leY6vut9VR4vSFUDE/W4sGjOSYZ086r7CkZ7Jpt0VZfJBA7wY9QM0YTzhfylF/X8SYE2MvOFlxHWJGyuwFT+sG1e0vaFbwei3eeOB9EhacZ0khDJEApMJv/EQYz4RjXMsv5lkc1/uY4ynJxxu+t9CVjN+FrmJMNSpZDXa4slyd5g8aoLLXFRg6oRs2faXMSO88GHI5+KRCodWEzmal2OfunmzWcNA2+aannsA3DuLzSClqi8+j8K34rFbT4vMsLddF9PDPkixNVitCfqv4N6FlxjNaEXNKOadzHbYueXkFVEcOClpU0M+yPNcUYnz7uZJHodCLaub5Kqli3RC/LBkZ1kt7SEghlCb6Ee4G0fcSOgQalyGSCvO8jT9QYZW2Yw84fg98ZJOsG4EWPgrctkKDobdPo+0y3Nmh2M6bMdzZx3Chnu+b5Q48Jcvd/xnL0Q6Ww7diOQRhL/CKLK+DLnCOxvfIjHAX8iGyjXzd9ZoxvG41kCmbmb2ovGgUaB6GHK+zi82I/zWvgdCM13cEx8Lyyz3BLEp/NWhwhonohq1WMlEYQEfXtcFRmehKqaM26JWjD2AlTx2Z5YQMTgtSxFmRvPMAFZ4yQAU2FdRurLFeH+ym/OYkRaXBCn81RCO5stcW2tfKpDGPp8nkweuyJauBIupFVvWBRS11AUR6BDtOS+2qZWhs073QSmGZMBqRshRonaVSPlP2SNgDo/UuW9DJQ+pVry+dNJ1l0gSrj43YGbF624h+Uv6qonL0nN8V6fUpaip1SnAmZ73mycEdmdOny4gKNjNsgJCv0fwKnlFY8M19uo4eP8Ro5/lOb2LkGZtwnqiLtA5PcWkJcnfUzUyDgyD3veMhDzyb+n+iwxv1I27fpYwHTlPKdOfxf1zJ7FuWtUpmZLihqmRANYsDLieEKX5bCWEIonMqZYL+cjfYSd7e+rhWebz2gyfoRQVe67yjV/luV+o9y2e7QBXyl3gKEVjNN6HraVgfm2/qC1LpkaP3t+PYrrGJ8oziPkpJvMzJe8lGrzrZqHe6ZBQC82Gr80+NQuc0qZFyVXXCB/rJjDpy6aBXqata1b7me27Cnhib5YuV31PPKzYrymmx2ToND3db9YDaJUbADWXshMDRSD9yOafsVdHGDHJfv3+fVG+4iN9vpDvluUt4iYcAPT2t8dNqe/PbRc+HAO5Pgte+ddmLXmYWejlCGtqs+yFyHQ1r9RzHeQip+fyHFNI7ElEWlx+SKqNf0JukisvtXz6aPdz+rwbd/gs=</diagram></mxfile>
```

Comment:
a few things about this graph:

1. Perhaps make it obvious that the `DB` here is the local db? Also add something to represent the customer's backend?
2. there should be something to indicate that the `Sync Processor` will perform the sync with the backend, as well as updating the local DB.
3. At the top of the graph, the `Sync Scheduler` has a line to `Update/Remove` records in DB. I think it means to update/remove the dataset clients. But I am wondering if it's actually required in this graph. I think it might be go into the implementation details. I think architecture graph should really about the big picture. 
4. At the bottom of the graph, instead of of `update` line to the DB, I think it's better to have a `Read` line from the db to make it obvious that it will use local data whenever possible?